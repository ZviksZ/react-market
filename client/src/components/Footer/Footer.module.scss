.footer {
    width: 100%;
    padding: 15px 0;
    margin-top: auto;
    background: #d1d1d1;
    box-shadow: 4px 4px 10px 2px rgba(0,0,0,0.2), 5px 4px 5px 10px rgba(0,0,0,0.14), 10px 10px 10px 10px rgba(0,0,0,0.12);

}
