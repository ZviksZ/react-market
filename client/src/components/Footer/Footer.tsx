import * as React from 'react'
import s from './Footer.module.scss'
import { Typography, Link } from '@material-ui/core'

export const Footer: React.FC = () => {
	return (
		<>
			<div className={s.footer}>
				<Typography variant="body2" color="textSecondary" align="center">
					{'Copyright Â© '}
					<Link color="inherit" href="https://react-e-commerce.com/">
						React E-commerce
					</Link>{' '}
					<span className="date">{new Date().getFullYear()}</span>
					{'.'}
				</Typography>
			</div>
		</>
	)
}
