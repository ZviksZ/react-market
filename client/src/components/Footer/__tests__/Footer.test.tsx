// @ts-nocheck
import * as React from 'react'
import { render } from '@testing-library/react'
import { Footer } from '../Footer'

describe('testing Footer component', () => {
	it('should be one link', () => {
		const { container } = render(<Footer />)
		const link = container.querySelectorAll('a')

		expect(link).toHaveLength(1)
	})

	it('should be current year', () => {
		const { container } = render(<Footer />)
		const date = container.querySelector('.date')
		const currentYear = new Date().getFullYear() + ''

		expect(date.textContent).toBe(currentYear)
	})

	it('hasn`t got changes', () => {
		const { container } = render(<Footer />)

		expect(container).toMatchSnapshot()
	})
})
