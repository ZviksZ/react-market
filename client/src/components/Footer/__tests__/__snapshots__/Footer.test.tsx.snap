// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Footer component hasn\`t got changes 1`] = `
<div>
  <div
    class="footer"
  >
    <p
      class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
    >
      Copyright Â© 
      <a
        class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
        href="https://react-e-commerce.com/"
      >
        React E-commerce
      </a>
       
      <span
        class="date"
      >
        2020
      </span>
      .
    </p>
  </div>
</div>
`;
